lib/services/role.go

```diff
@@ -2035,25 +2048,25 @@ func (set RoleSet) checkAccess(r AccessCheckable, mfa AccessMFAParams, matchers
 
 		matchLabels, labelsMessage, err := MatchLabels(getRoleLabels(role, types.Deny), allLabels)
 		if err != nil {
-			return trace.Wrap(err)
+			return predicateEngine.AccessUndecided, trace.Wrap(err)
 		}
 		if matchLabels {
 			debugf("Access to %v %q denied, deny rule in role %q matched; match(namespace=%v, label=%v)",
 				r.GetKind(), r.GetName(), role.GetName(), namespaceMessage, labelsMessage)
-			return trace.AccessDenied("access to %v denied. User does not have permissions. %v",
+			return predicateEngine.AccessUndecided, trace.AccessDenied("access to %v denied. User does not have permissions. %v",
```

Comment:
isn't this an explicit deny? why does it return undecided?

Reply:
Good catch, thanks